src/ensembl/src/shared/hooks/useOutsideClick.tsx

```diff
@@ -0,0 +1,21 @@
+import { useEffect } from 'react';
+
+export default function useOutsideClick<T extends HTMLElement>(
+  ref: React.RefObject<T>,
+  callback: () => void
+): void {
+  const handleClickOutside = (event: Event) => {
+    if (ref.current && !ref.current.contains(event.target as HTMLElement)) {
+      callback();
+    }
+  };
+
+  useEffect(() => {
+    // Bind the event listener
+    document.addEventListener('click', handleClickOutside);
+    return () => {
+      // Unbind the event listener on clean up
+      document.removeEventListener('click', handleClickOutside);
```

Comment:
This function too (like above) can be written as a one liner.

Reply:
That's done